Sources/KlaviyoSwift/StateManagement/StateManagement.swift

```diff
@@ -106,22 +107,30 @@ enum KlaviyoAction: Equatable {
     }
 }
 
-struct RequestId {}
-struct FlushTimer {}
+enum CancelIds {
+    case request
+    case timer
+}
 
-struct KlaviyoReducer: ReducerProtocol {
+struct KlaviyoReducer: Reducer {
     typealias State = KlaviyoState
     typealias Action = KlaviyoAction
 
-    func reduce(into state: inout KlaviyoState, action: KlaviyoAction) -> EffectTask<KlaviyoAction> {
+    var body: any Reducer {
+        Reduce { state, action in
+            reduce(into: &state, action: action)
+        }
+    }
+
+    func reduce(into state: inout State, action: Action) -> Effect<Action> {
         if action.requiresInitialization,
            case .uninitialized = state.initalizationState {
-            environment.emitDeveloperWarning("SDK must be initialized before usage.")
+            environment.logger.error("SDK must be initialized before usage.")
```

Comment:
This requires access to mainactor and cannot be async currently so it has been switch to a logger error instead. This is done in many places.